### ISSUE TITLE
howtogeek.com is almost unusable as NVDA fails to detect links in Browse mode

### ISSUE THREAD

**Issue by amirsol81** [nvaccess/nvda#11709]
On www.howtogeek.com NVDA doesn't detect most of the links and, as a consequence, fails to activate them unless Focus mode is used. This is with Chrome/MS Edge 85.
### Steps to reproduce:
1. Using MS Edge 85 or Google Chrome 85, open www.howtogeek.com
2. Navigate by heading. Each heading is actually the title of an article, and each heading actually contains a link. But NVDA fails to detect the links embedded in headings in Browse mode which is the default mode for HTML content.
### Actual behavior:
In Browse mode, links embedded in headings aren't detected and announced, and pressing Enter doesn't activate them either. Link detection works in Focus mode, but it's quite cumbersome.
### Expected behavior:
NVDA should detect, announce and activate the links on each heading in Browse mode.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
Alpha-21082 and alphas going back to a few weeks. Also the official 2020.2, and the latest 2020.3 beta
#### Windows version:
Win 10 Pro 64-Bit, Version 2004
#### Name and version of other software in use when reproducing the issue:
Google Chrome 85 and MS Edge 85
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Mentioned above: latest alpha/stable/beta
#### If addons are disabled, is your problem still occuring?
Yes
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No.

**Comment by amirsol81** [comment]
It's worth mentioning that JAWS 2020 (latest monthly release) works flawlessly on www.howtogeek.com

**Comment by feerrenrut** [comment]
This looks like an author error on the howtogeek page, here is a minimal test case based on their site:
`data:text/html, <div><p>text before</p><header aria-label="This is the aria-label text" role="heading"><h2><a title="anchor title text" href="">The displayed text</a></h2></header><div>`

We could be presenting this differently, but that would break other pages. It might be possible for us to work around this, the downside is that it doesn't discourage bad practices on the web.

Let's unpack what they actually have on their site a bit. I've removed some irrelevant attributes and tags.
``` html
<header aria-label="This Is the Most Popular Classic TV Show in Your State" role="heading">
    <h2>
         <a title="This Is the Most Popular Classic TV Show in Your State" href="">This Is the Most Popular Classic TV Show in Your State</a
    </h2>
</header>
```

Compare that to the following which is announced as expected: `data:text/html, <div><p>text before</p><h2><a href="">an article title</a></h2><div>`

**Comment by feerrenrut** [comment]
I'd like to determine if there has been a regression. Is there an older version of NVDA that works as you expect with this sample?

**Comment by amirsol81** [comment]
> This looks like an author error on the howtogeek page, here is a minimal test case based on their site:
> `data:text/html, <div><p>text before</p><header aria-label="This is the aria-label text" role="heading"><h2><a title="anchor title text" href="">The displayed text</a></h2></header><div>`
> 
> We could be presenting this differently, but that would break other pages. It might be possible for us to work around this, the downside is that it doesn't discourage bad practices on the web.
> 
> Let's unpack what they actually have on their site a bit. I've removed some irrelevant attributes and tags.
> 
> ```
> <header aria-label="This Is the Most Popular Classic TV Show in Your State" role="heading">
>     <h2>
>          <a title="This Is the Most Popular Classic TV Show in Your State" href="">This Is the Most Popular Classic TV Show in Your State</a
>     </h2>
> </header>
> ```
> 
> Compare that to the following which is announced as expected: `data:text/html, <div><p>text before</p><h2><a href="">an article title</a></h2><div>`

@feerrenrut Thanks for the analysis. Given the relative popularity of the website, any chance of getting NVDA to work with it similar to what JAWS does? Guess JAWS is more tolerant of such an error.

**Comment by amirsol81** [comment]
> I'd like to determine if there has been a regression. Is there an older version of NVDA that works as you expect with this sample?

Not that I can remember. At least the previous 2 stable releases don't.

**Comment by feerrenrut** [comment]
> At least the previous 2 stable releases don't.

It might be that the website has changed. We treat regressions slightly differently, so it is helpful to know. Given the last two versions behave in the same way, then we won't treat this as a regression.

>getting NVDA to work with it similar to what JAWS does? Guess JAWS is more tolerant of such an error.

It's hard to say without spending more time thinking about the implications. It might be hard to avoid causing duplicate speech in other situations, or potentially incorrect semantics, and JAWS may have these problems on other sites. 

The pattern being used on this sample seems to be intentionally trying to alter the semantics of the attribute, one might even say that it is a bug in JAWS that link is announced.

Looking at `role=heading` in section [5.3.4 Accessible Name Guidance by Role](https://www.w3.org/TR/wai-aria-practices-1.1/#naming_role_guidance) of the WAI-ARIA Authoring Practices guides, it says:
>heading - Required Only If Content Insufficient - Warning! Using aria-label or aria-labelledby will hide descendant content from assistive technologies. Ideally named by visible, descendant content.

**Comment by amirsol81** [comment]
This issue has been resolved, so I'm closing it.
